scripts/src/main/resources/scripts/command/npm

```diff
@@ -16,13 +16,47 @@ source "$(dirname "${0}")"/../functions
 
 # $1: optional setup
 function doSetup() {
-  if [ ! -x "${NPM_CMD}" ]
+  local software_version=${NPM_VERSION}
+  local installed_version
+  if [ -x "${NPM_CMD}" ]
   then
+    installed_version=$(${NPM_CMD} -version)
+  else 
     doDevonCommand node setup silent
-  elif [ "${1}" != "silent" ]
-  then
-    doEcho "npm is already installed at ${NPM_CMD}"
+    installed_version=$(${NPM_CMD} -version)
   fi
+
+  if [ -z "$software_version" ] || [ "$installed_version" == "$software_version" ] 
+  then
+    if [ "${1}" != "silent" ]
+    then
+      doEcho "Version ${installed_version} of npm is already installed at ${NPM_CMD}"
+    fi
+  else
+    local download_url="https://registry.npmjs.org/npm/-/npm-${software_version}.tgz"
+    local software_dir="${NODE_HOME}/node_modules/npm"
+    doInstall "${software_dir}" "${download_url}" "npm" "${software_version}"
```

Comment:
this will not work as you have developed this PR for an older version of devonfw-ide and the function and download process has been improved and refactored. See: #641 
So we should to put the URL into https://github.com/devonfw/ide-mirrors/ and change the arguments of `doInstall` (see other commandlets that have been refactored already).

Reply:
As a quickfix, I changed the order of the parameters so it should already work, but we want to keep the download URLs out of our release as explained in #641

Reply:
I have already added the according config to the ide-mirrors repo:
https://github.com/devonfw/ide-mirrors/tree/master/npm

Reply:
so we could stip out the download_url (line 36) and change to:
```
doInstall "-" "${software_dir}" "npm" "${software_version}"
```